src/mlagility/api/model_api.py

```diff
@@ -116,10 +123,71 @@ def benchmark_model(
                 )
                 perf = cpu_model.benchmark(backend=backend)
 
+        elif device == "x86" and runtime in ["torch", "torch_compiled"]:
+
+            # Create cache folder with stats file
+            # Although building with an empty sequence is possible, we don't want
+            # the model to be rebuilt when other devices are used, causing the
+            # results inside of the stats file to be lost.
+            build_model(
+                model=model,
+                inputs=inputs,
+                build_name=build_name,
+                cache_dir=cache_dir,
+                rebuild=rebuild,
+                sequence=sequence,
+            )
+
+            torch_version = torch.__version__
+            if runtime == "torch_compiled":
+                clean_torch_version = torch_version.split("+")[0]
+                if version.parse(clean_torch_version) < version.parse("2.0.0"):
+                    BenchmarkException(
+                        (
+                            f"{runtime} can only be used with Pytorch 2.0.0 or above. "
+                            f"However, version {torch_version} was found."
+                        )
+                    )
+
+                model = torch.compile(model)
```

Comment:
`benchmark_scripts()` can loop over the same `model` multiple times, for example if the `devices` argument has more than one member (e.g., `devices=["x86","nvidia"]`). 

What happens when the user does `devices=["x86","nvidia"]` with `runtime="torch_compiled"`? Is this line of code a change to the `model` object that will impact other calls to `benchmark_model()`?

Put another way, should you be making a deepcopy of `model` before running `torch.compile(model)`?

Reply:
PS. it also occurs to me that we might want to just change the `devices` arg to `device` (singular). The only reason it's `devices` was to maintain feature-for-feature compatibility with `autogroq`.

However even if we do that, you still may want to make the `deepcopy`. After all, who knows what user code plans to do with `model` after this function runs?

Reply:
Not reassigning function arguments anymore!